sdk/include/stdlib.h

```diff
@@ -145,6 +145,27 @@ void *__cheri_compartment("alloc")
 size_t __cheri_compartment("alloc")
   heap_claim(struct SObjStruct *heapCapability, void *pointer);
 
+/**
+ * Interface to the fast claims mechanism.  This claims two pointers using the
+ * hazard-pointer-inspired lightweight claims mechanism.  If this function
+ * returns zero then the heap pointers are guaranteed not to become invalid
+ * until either the next cross-compartment call or the next call to this
+ * function.
+ *
+ * A null pointer can be used as a not-present value.  This function will
+ * treat operations on null pointers as unconditionally successful.  It returns
+ * `-ETIMEDOUT` if it failed to claim before the timeout expired or `EINVAL` if
+ * one or more of the arguments is neither null nor a valid pointer at the end.
+ *
+ * In the case of failure, neither pointer will have been claimed.
+ *
+ * This function is provided by the compartment_helpers library, which must be
+ * linked for it to be available.
+ */
```

Comment:
Is the idiomatic way to release a claim to call with `nullptr`?

Reply:
Yup.  Or simply return from a cross-compartment call (or call another cross-compartment call).  For example, if you're yielding in a loop, you implicitly release them before each loop iteration, so you don't need the explicit call on retry.